src/lib/paymentOptions/NimiqPaymentOptions.ts

```diff
@@ -0,0 +1,186 @@
+import { TX_VALIDITY_WINDOW, TX_MIN_VALIDITY_DURATION } from '../Constants';
+import { Currency, PaymentMethod, PaymentOptions } from '../PublicRequestTypes';
+import { ParsedPaymentOptions } from '../RequestTypes';
+import { toNonScientificNumberString } from '@nimiq/utils';
+
+export interface NimiqProtocolSpecific {
+    fee?: number | string;
+    feePerByte?: number | string;
+    validityDuration?: number;
+    flags?: number;
+    sender?: string;
+    forceSender?: boolean;
+    recipient?: string;
+    recipientType?: Nimiq.Account.Type;
+}
+
+export type ParsedNimiqProtocolSpecific = Omit<NimiqProtocolSpecific, 'sender' | 'recipient' | 'fee' | 'feePerByte'>
+    & Required<Pick<NimiqProtocolSpecific, 'forceSender' | 'flags' | 'validityDuration'>> & {
+    sender?: Nimiq.Address,
+    recipient?: Nimiq.Address,
+    fee: number,
+    feePerByte: number,
+};
+
+export type NimiqDirectPaymentOptions = PaymentOptions<Currency.NIM, PaymentMethod.DIRECT>;
+
+export class ParsedNimiqDirectPaymentOptions extends ParsedPaymentOptions<Currency.NIM, PaymentMethod.DIRECT> {
+    public amount: number;
+    private extraData: Uint8Array;
+
+    public constructor(options: NimiqDirectPaymentOptions, extraData: Uint8Array) {
+        super(options);
+        this.extraData = extraData;
+        this.amount = Number.parseInt(toNonScientificNumberString(options.amount), 10);
+
+        let sender: Nimiq.Address | undefined;
+        if (options.protocolSpecific.sender !== undefined) {
+            try {
+                sender = Nimiq.Address.fromUserFriendlyAddress(options.protocolSpecific.sender);
+            } catch (err) {
+                throw new Error('If provided, sender must be a valid user friendly address string');
+            }
+        }
+
+        let recipient: Nimiq.Address | undefined;
+        if (options.protocolSpecific.recipient !== undefined) {
+            try {
+                recipient = Nimiq.Address.fromUserFriendlyAddress(options.protocolSpecific.recipient);
+            } catch (err) {
+                throw new Error('If provided, recipient must be a valid user friendly address string');
+            }
+        }
+
+        let recipientType: Nimiq.Account.Type | undefined;
+        if (options.protocolSpecific.recipientType !== undefined) {
+            if (!Object.values(Nimiq.Account.Type).includes(options.protocolSpecific.recipientType)) {
+                throw new Error('If provided, recipientType must be a valid nimiq account type');
+            }
+            recipientType = options.protocolSpecific.recipientType;
+        }
+
+        const flags = (options as NimiqDirectPaymentOptions).protocolSpecific.flags;
```

Comment:
Type casting should be unnecessary as the argument to the constructor is already typed.